core/reader/replicate_channel_manager.go

```diff
@@ -174,71 +178,96 @@ func (r *replicateChannelManager) StartReadCollection(ctx context.Context, info
 
 	sourceDBInfo := r.metaOp.GetDatabaseInfoForCollection(ctx, info.ID)
 
+	var shouldSendAPIEvent bool
 	var err error
-	retryErr := retry.Do(ctx, func() error {
-		_, err = r.targetClient.GetCollectionInfo(ctx, info.Schema.GetName(), sourceDBInfo.Name)
-		if err != nil &&
-			!IsCollectionNotFoundError(err) && !IsDatabaseNotFoundError(err) {
-			return err
-		}
-		r.collectionLock.RLock()
-		_, ok := r.replicateCollections[info.ID]
-		r.collectionLock.RUnlock()
-		if ok {
-			return errors.Newf("the collection has been replicated, wait it [collection name: %s] to drop...", info.Schema.Name)
+	if r.downstream == "milvus" {
+		retryErr := retry.Do(ctx, func() error {
+			_, err = r.targetClient.GetCollectionInfo(ctx, info.Schema.GetName(), sourceDBInfo.Name)
+			if err != nil &&
+				!IsCollectionNotFoundError(err) && !IsDatabaseNotFoundError(err) {
+				return err
+			}
+			r.collectionLock.RLock()
+			_, ok := r.replicateCollections[info.ID]
+			r.collectionLock.RUnlock()
+			if ok {
+				return errors.Newf("the collection has been replicated, wait it [collection name: %s] to drop...", info.Schema.Name)
+			}
+			// collection not found will exit the retry
+			return nil
+		}, r.startReadRetryOptions...)
+
+		if retryErr != nil {
+			return retryErr
 		}
-		// collection not found will exit the retry
-		return nil
-	}, r.startReadRetryOptions...)
 
-	if retryErr != nil {
-		return retryErr
+		if err != nil {
+			// the collection is not existed in the target and source collection has dropped, skip it
+			if info.State == pb.CollectionState_CollectionDropped || info.State == pb.CollectionState_CollectionDropping {
+				r.droppedCollections.Store(info.ID, struct{}{})
+				log.Info("the collection is dropped in the target instance",
+					zap.Int64("collection_id", info.ID), zap.String("collection_name", info.Schema.Name))
+				return nil
+			}
+			if IsDatabaseNotFoundError(err) {
+				log.Panic("the database has been dropped but the collection is existed in the source instance",
+					zap.String("collection_name", info.Schema.Name))
+				return nil
+			}
+			select {
+			case <-ctx.Done():
+				log.Warn("context is done in the start read collection")
+				return ctx.Err()
+			default:
+				shouldSendAPIEvent = true
+			}
+		}
+	} else {
+		shouldSendAPIEvent = true
```

Comment:
Why the `shouldSendAPIEvent` is true when the downstream isn't milvus. If this, every time the cdc is started, or paused and resumed, an API event is sent, which seems to be a problem